cmd/caa-log-checker/main.go

```diff
@@ -79,6 +84,13 @@ func checkIssuances(scanner *bufio.Scanner, checkedMap map[string][]time.Time, t
 			return fmt.Errorf("line %d: failed to parse timestamp: %s", lNum, err)
 		}
 
+		if !checkFromDay.IsZero() && !checkUntilDay.IsZero() &&
```

Comment:
This logic checks that both flags are non-zero and does nothing otherwise. We should verify during flag parsing that they are either both unset or both set.

Also I think `earliest` and `latest` are slightly clearer time words and are used elsewhere in Boulder (e.g. in the SA).

Reply:
We do check during flag parsing that both flags are either present or absent. Are you saying this check could be simplified to check whether either date `IsZero()`? This function does need to handle the case where no `-check-from` and `-check-until` are provided, i.e. the case where the Boulder with this change is deployed ahead of the enclosing script that adopts the new parameters.

re `earliest`/`latest`: I'm happy to go with the more consistent terminology, though I wonder if `latest` captures the fact that it's an exclusive boundary. If we change `-check-from` and `-check-until` to `-earliest` and `-latest`, and I pass `-earliest 20200818 -latest 20200820`, I'd be surprised when issuances from 2020-08-20 are excluded from the check. I think `until` makes it a little more clear.

Reply:
> We do check during flag parsing that both flags are either present or absent.

You're right, I misunderstood the code. It looks like it will fail with a parse error if one is set but the other is unset. I would lean in favor of a more explicit check:

```
	if *checkFromFlag != "" || *checkUntilFlag != "" {
            if *checkFromFlag == "" || checkUntilFlag == "" {
                cmd.Fail("both -check-from and -check-until must be both set or both unset")
            }
```

Reply:
Re: earliest versus latest, my reason for avoiding "from" is that it's ambiguous in two ways:
 - "From" can mean the source of something like "where did this come from" in addition to a range "check from time X to time Y".
 - "From" is ambiguous as to direction. Do we start at the "from" date and work backward or work forward? Speaking of which, we should add a check that these two flags have the correct time relationship to each other, i.e. one after the other.

Earliest and latest also read nicely with Go's built in time comparison operators:

```
(ie.issuanceTime.Before(earliest) || ie.issuanceTime.After(latest))
```